java-security/src/test/java/com/sap/cloud/security/adapter/xs/XSUserInfoAdapterTest.java

```diff
@@ -521,6 +521,46 @@ public void isForeignModeTrue_whenIdentityZoneIsMissing() throws XSUserInfoExcep
 		assertThat(cut.isInForeignMode()).isTrue();
 	}
 
+	@Test
```

Comment:
Here's a lot of repeated code, I would suggest to create these test cases as parameterized test and use Mocking for XsuaaToken instead of creating multiple Jwt token files in resources.